### ISSUE TITLE
method_cache causes TypeError when deepcopying object

### ISSUE THREAD

**Issue by jaraco** [jaraco/jaraco.functools#1]
Attempting to deepcopy an object that has a `method_cache` wrapped method which has been invoked at least once results in a TypeError when deepcopy attempts to copy the cache.

----------------------------------------
- Bitbucket: https://bitbucket.org/jaraco/jaraco.functools/issue/1
- Originally reported by: Jason R. Coombs
- Originally created at: 2015-10-20T16:24:28.089

**Comment by jaraco** [comment]
Add test capturing failure where a deepcopy fails after a method has been cached. Ref #1.

→ <<cset 68d1fda21af7>>

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
It seems the issue can be simply recreated with

```
$ cat >> cache_copy.py
import copy
from functools import lru_cache
foo = lru_cache()(lambda: None)
copy.deepcopy(foo)
$ python3.5 cache_copy.py
Traceback (most recent call last):
  File "cache_copy.py", line 4, in <module>
    copy.deepcopy(foo)
  File "/usr/lib/python3.5/copy.py", line 182, in deepcopy
    y = _reconstruct(x, rv, 1, memo)
  File "/usr/lib/python3.5/copy.py", line 293, in _reconstruct
    y = callable(*args)
  File "/usr/lib/python3.5/copyreg.py", line 88, in __newobj__
    return cls.__new__(cls, *args)
TypeError: Required argument 'user_function' (pos 1) not found
```

The issue only occurs on Python 3.5. The same execution succeeds on Python 3.4.

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Bug [filed upstream](http://bugs.python.org/issue25447).

----------------------------------------
Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
This error issue was fixed in Python 3.5.1, although a subsequent issue emerged as #3.
